packages/webpack/src/BuildMetadataPlugin.ts

```diff
@@ -0,0 +1,85 @@
+import * as path from 'node:path';
+import { PackageJson } from 'type-fest';
+import { type WebpackPluginInstance, BannerPlugin, Compilation, Compiler, sources } from 'webpack';
+
+import { createBannerComment, getBuildMetadata } from '@kubev2v/build/src/metadata';
+
+export type BuildMetadataPluginOptions = {
+  /**
+   * The contents of `package.json` to provide package name and version.
+   */
+  packageJson: PackageJson;
+
+  /**
+   * Path and name of the JSON metadata file, relative to the configured webpack output path.
+   *
+   * Default value: `build-metadata.json`
+   */
+  jsonMetadataFilename?: string;
+
+  /**
+   * The tag to include in the chunk banner.  This can be used to configure terser to
+   * retain the banner comment in the optimized chunks.
+   *
+   * Default value: `@buildInfo`
+   */
+  bannerBuildInfoTag?: string;
+};
+
+/**
+ * Collect and output build metadata from a provided `package.json` and from calls to
+ * git.  The metadata is saved in two places:
+ *   - A banner comment starting with `bannerBuildInfoTag` in each chunk
+ *   - A json metadata file
+ */
+export class BuildMetadataPlugin implements WebpackPluginInstance {
+  private readonly packageJson: PackageJson;
+  private readonly jsonMetadataFilename: string;
+  private readonly bannerBuildInfoTag: string;
+
+  constructor(options: BuildMetadataPluginOptions = { packageJson: {} }) {
+    this.packageJson = options.packageJson;
+    this.jsonMetadataFilename = options.jsonMetadataFilename ?? 'build-metadata.json';
+    this.bannerBuildInfoTag = options.bannerBuildInfoTag ?? '@buildInfo';
+  }
+
+  apply(compiler: Compiler) {
+    //
+    // Lookup the build metadata (package.json of the thing running webpack & git info):
+    //
+    const buildInfo = getBuildMetadata(this.packageJson);
+
+    //
```

Comment:
the banner const and the banner plugin should be in the forklift console plugin webpack config
one plugin to add the metadata file, and another plugin for the banner

can you move lines 52 .. 64 to the forklift console plugin package ?

Reply:
The webpack config now does all the build metadata work.  The new webpack plugin only outputs JSON to a file.